src/cursor/cur_table.c

```diff
@@ -837,18 +837,25 @@ __curtable_largest_key(WT_CURSOR *cursor)
 static int
 __curtable_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i, j;
 
+    i = 0;
     ctable = (WT_CURSOR_TABLE *)cursor;
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
-    /* Grab the primary cursor and call bound function. */
-    primary = *ctable->cg_cursors;
-    WT_ERR(primary->bound(primary, config));
+    /* Call bound function on all column groups. */
+    for (i = 0, cp = ctable->cg_cursors; i < WT_COLGROUPS(ctable->table); i++, cp++)
+        WT_ERR((*cp)->bound(*cp, config));
+
 err:
+    /* If applying bounds fails on one colgroup, reset all of them for consistency. */
+    if (ret != 0)
+        for (j = 0, cp = ctable->cg_cursors; j <= i; j++, cp++)
+            WT_IGNORE_RET((*cp)->bound(*cp, "action=clear"));
```

Comment:
I don't think we should ignore the return of this call, if it fails we'd be in a very bad state. If anything we should PANIC.

Reply:
[6b20d5f](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/6b20d5fcb91c96bb46167462ff7aedb3d0f9ccf0)